app/src/main/java/com/android/doctorapp/ui/doctor/AddDoctorViewModel.kt

```diff
@@ -0,0 +1,158 @@
+
+    val doctorContactNumber: MutableLiveData<String> = MutableLiveData()
+    val doctorContactNumberError: MutableLiveData<String?> = MutableLiveData()
+
+    val toggleLiveData: MutableLiveData<Boolean> = MutableLiveData(false)
+    private lateinit var firebaseAuth: FirebaseAuth
+    val isDataValid: MutableLiveData<Boolean> = MutableLiveData(false)
+
+    private val _navigationListener = SingleLiveEvent<Int>()
+    val navigationListener = _navigationListener.asLiveData()
+
+    private val _addDoctorResponse = SingleLiveEvent<String>()
+    val addDoctorResponse = _addDoctorResponse.asLiveData()
+
+    private fun isAllValidate() {
+        isDataValid.value = (!doctorName.value.isNullOrEmpty() && !doctorEmail.value.isNullOrEmpty()
+                && !doctorContactNumber.value.isNullOrEmpty() && doctorNameError.value.isNullOrEmpty()
+                && doctorEmailError.value.isNullOrEmpty() && doctorContactNumberError.value.isNullOrEmpty())
+    }
+
+    fun isValidateName(text: CharSequence?) {
+        if (text?.toString()
+                .isNullOrEmpty() || ((text?.toString()?.length
+                ?: 0) < 3)
+        ) {
+            doctorNameError.value = resourceProvider.getString(R.string.valid_name_desc)
+        } else {
+            doctorNameError.value = null
+
+        }
+        isAllValidate()
+    }
+
+    fun isValidEmail(text: CharSequence?) {
+        if (text?.toString().isNullOrEmpty() || text?.toString()?.isEmailAddressValid()
+                ?.not() == true
+        ) {
+            doctorEmailError.value = resourceProvider.getString(R.string.enter_valid_email)
+        } else {
+            doctorEmailError.value = null
+        }
+        isAllValidate()
+    }
+
+    fun isValidContact(text: CharSequence?) {
+        if (text?.toString().isNullOrEmpty()) {
+            doctorContactNumberError.value =
+                resourceProvider.getString(R.string.error_valid_phone_number)
+        } else {
+            if (Patterns.PHONE.matcher(text ?: "").matches()) {
+                if (text?.toString()?.length == 10) {
+                    doctorContactNumberError.value = null
+                } else {
+                    doctorContactNumberError.value =
+                        resourceProvider.getString(R.string.error_valid_phone_number)
+                }
+            } else {
+                doctorContactNumberError.value =
+                    resourceProvider.getString(R.string.error_valid_phone_number)
+            }
+        }
+        isAllValidate()
+    }
+
+
+    fun addDoctorData() {
+        // Initialize firebase auth
+        firebaseAuth = FirebaseAuth.getInstance()
+        val firestore = FirebaseFirestore.getInstance()
+
+        // Initialize firebase user
+        val firebaseUser = firebaseAuth.currentUser
+        if (firebaseUser != null) {
+            // when firebaseUser is not null then
+            val userData = UserDataRequestModel(
```

Comment:
User default argument in data class. After that no need to define each and every field